drake/solvers/NonlinearProgramSnoptmex.cpp

```diff
@@ -16,6 +17,14 @@ using namespace std;
 
 static char* snopt_userfun_name;
 
+// NOTE: all snopt calls will use this shared memory... so this code is NOT THREAD SAFE
+static unique_ptr<snopt::doublereal []> rw;
+static unique_ptr<snopt::integer []> iw;
+static unique_ptr<char []> cw;
+static snopt::integer lenrw=0;
+static snopt::integer leniw=0;
+static snopt::integer lencw=0;
+
 static int snopt_userfun(snopt::integer *Status, snopt::integer *n, snopt::doublereal x[],
```

Comment:
+1 for making them static. 

What is our preference in using pointer? Should we always prefer smart pointer to raw pointer?